src/EVM/SMT.hs

```diff
@@ -788,12 +790,14 @@ exprToSMT = \case
   EqByte a b ->
     let cond = op2 "=" a b in
     "(ite " <> cond `sp` one `sp` zero <> ")"
-  Keccak a ->
-    let enc = exprToSMT a in
-    "(keccak " <> enc <> ")"
-  SHA256 a ->
-    let enc = exprToSMT a in
-    "(sha256 " <> enc <> ")"
+  Keccak a -> let
+      enc = exprToSMT a
+      sz = exprToSMT $ Expr.simplify $ BufLength a
```

Comment:
Shouldn't we use `bufLength` function from `Expr` here?
We do that when computing injectivity constraints in `injProp` in `Keccak`.
Feels like these should be unified...

Reply:
Ah good point. The `Expr.simplify $BufLength x` effectively does **something** like that. If you look at `Expr.simplify`, line 957:

```haskell
    go (BufLength buf) = bufLength buf
```

it actually calls `Expr.bufLength`. However, `Expr.simplify $BufLength x` calls simplify on `x` as well, recursively.

However, `x` should have been simplified at this point, and should need no simplifixation. And also, we should NOT simplify things here -- some of our fuzzing systems rely on simplification NOT taking place:

See line 3650 in `test.hs`:

```haskell
     let smt = assertPropsNoSimp conf [mkprop l r]
```

If simplification takes place, then it's not possible to check simplified vs unsimplified expressions against each other. Thereby making it impossible to check for simplification bugs.

So I am replacing this with `Expr.bufLength`. Thanks for spotting this!

Reply:
I am closing. Please do reopen in case you think my fix is wrong :)